sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/TestMethodShouldHaveCorrectSignature.Misc.cs

```diff
@@ -0,0 +1,57 @@
+﻿namespace Tests.Diagnostics
+{
+    using System.Threading.Tasks;
+    using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+    public partial class TestThatIssuesAreOnlyReportedOnceForPartialClasses
+    {
+    }
+
+    [TestClass]
+    public partial class TestThatIssuesAreOnlyReportedOnceForPartialClasses
```

Comment:
The previous version would report each issue multiple times for partial classes since it was starting at the _ClassDeclarationSyntax_ node, then calling _GetMethods_ on the _ClassSymbol_.